crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1444,7 +1445,7 @@ impl<'db> TypeInferenceBuilder<'db> {
 
         let expr_id = expression.scoped_ast_id(self.db, self.scope);
         let previous = self.types.expressions.insert(expr_id, ty);
-        assert!(previous.is_none());
+        assert_eq!(previous, None);
```

Comment:
(This gets you a much better error message as it tells you what `previous` was in the error message)